integration/simulation/transaction_injector.go

```diff
@@ -38,6 +39,9 @@ type TransactionInjector struct {
 
 	l2TransactionsLock sync.RWMutex
 	l2Transactions     enclave.L2Txs
+
+	running bool
```

Comment:
wouldn't a single flag be enough here?
Also, you have to use some synchronisation primitive to make sure this works. We've used atomic load and set everywhere else for stuff like this

Reply:
A single flag might not cut it, it uses 3 states: Running, NotRunning, IsStopped. And NotRunning doesn't mean it's IsStopped - A transaction might still be going through. That was my thinking.

Do we need sync ? There's one writer, many readers and the values only go in one direction. 
Is there a case here that I'm not seeing that it might bork 🤔  ?
Happy to add sync otherwise.

Reply:
> A single flag might not cut it, it uses 3 states: Running, NotRunning, IsStopped. And NotRunning doesn't mean it's IsStopped - A transaction might still be going through. That was my thinking.
> 
Feels like it's a bit overthinking it. When you call "Start", you start injecting txs, and when you call stop you signal for it to stop injecting any further txs.

> Do we need sync ? There's one writer, many readers and the values only go in one direction. Is there a case here that I'm not seeing that it might bork 🤔 ? Happy to add sync otherwise.

Pretty much says it here: https://go.dev/ref/mem
_"Programs that modify data being simultaneously accessed by multiple goroutines must serialize such access."_

Reply:
"Don't be clever" hahaha ! 

Yeah, best practices ARE best 👍 

Refactored :)